### PR TITLE
First iteration of Prot Warrior - Looking for feedback

### PR DESCRIPTION
**Pull request by cheesypeaz**
- Based upon BFA spec ability priority
- Includes option to include defensive and mitigation abilities
- Supports inline and off GCD concepts 
- Option/Setting to use Last Stand as a filler during Shield Block down time (Currently working on a more optimal version of this)
- Supports Vengence however I do not currently use this
- Incorporates Free Revenge Procs 
- Has a simple mechanism to prevent Rage cap based upon whether player is currently tanking or not

### CODE REVIEW

**Comment by aethys256** [comment]
Thanks, looking good at a first glance, I'll review by tonight.
Maybe you wanna port the logic into a SimC APL so it'll be easier to improve it (and generate all this automatically from the generator with ease once it'll be fully ready).

File: HeroRotation_Monk/Brewmaster.lua [hunk 1]
```diff
@@ -228,4 +228,4 @@ local function APL()
   end
 end
 
-HR.SetAPL(268, APL)
```
Comment (aethys256) [inline HeroRotation_Monk/Brewmaster.lua]: Guess you removed the final line return, you should avoid doing so.
Every file should end with a line return ^^

File: HeroRotation_Warrior/Protection.lua [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+local Item = HL.Item;
+
+-- HeroRotation
+local HR = HeroRotation;
+
+-- APL Local Vars
+local Everyone = HR.Commons.Everyone;
+-- Spells
+if not Spell.Warrior then Spell.Warrior = {}; end
+Spell.Warrior.Protection = {
```
Comment (aethys256) [inline HeroRotation_Warrior/Protection.lua]: In general it's good to align ` = Spell(...)` there for readability.
You can use plugin to do that automatically. like the "Align" extension on VSCode.

File: HeroRotation_Warrior/Protection.lua [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  -- is player currently tanking any enemies within 16 yard radius
+  local IsTanking = Player:IsTankingAoE(16) or Player:IsTanking(Target);
+  return IsTanking;
+end
+
+-- Determine if it is optimal to cast ip. We dont want to waste rage and over cap it
+local function shouldCastIp()
+
+  if Player:Buff(S.IgnorePain) then 
+    local castIP = tonumber((GetSpellDescription(190456):match("up to (.-) total damage prevented."):gsub(",","")));
```
Comment (aethys256) [inline HeroRotation_Warrior/Protection.lua]: Won't work for non-english client. You might want to go with a more solid approach than tooltip reading.
My advice would be to recompute the value using the formula, you have some example in the Rogue module.
See `S.Eviscerate:RegisterDamage()`for example.

Comment (cheesypeaz) [reply HeroRotation_Warrior/Protection.lua]: ill Fix in next version. I had the formula originally but it does not match real values. Wowhead formula nor warrior discord seems correct right now. Still looking into it.

Comment (cheesypeaz) [reply HeroRotation_Warrior/Protection.lua]: ill Fix in next version. I had the formula originally but it does not match real values. Wowhead formula nor warrior discord seems correct right now. Still looking into it.

File: HeroRotation_Warrior/Protection.lua (hunk 2 above)
Comment (cheesypeaz) [reply HeroRotation_Warrior/Protection.lua]: I’ll fix

File: HeroRotation_Monk/Brewmaster.lua (hunk 1 above)
Comment (cheesypeaz) [reply HeroRotation_Monk/Brewmaster.lua]: Sry.. that was a merge problem I missed

**Pull request merged**